packages/flutter/test/material/paginated_data_table_test.dart

```diff
@@ -1043,4 +1044,81 @@ void main() {
     await tester.pumpWidget(buildFrame(overflowBar));
     expect(headerX, tester.getTopLeft(find.byType(ElevatedButton)).dx);
   });
+
+  testWidgets('PaginatedDataTable can be scrolled using ScrollController', (WidgetTester tester) async {
+    final TestDataSource source = TestDataSource();
+    final ScrollController scrollController = ScrollController();
+
+    Widget buildTable(TestDataSource source) {
+      return Align(
+        alignment: Alignment.topLeft,
+        child: SizedBox(
+          width: 100,
+          child: PaginatedDataTable(
+            controller: scrollController,
+            header: const Text('Test table'),
+            source: source,
+            rowsPerPage: 2,
+            columns: const <DataColumn>[
+              DataColumn(
+                label: Text('Name'),
+                tooltip: 'Name',
+              ),
+              DataColumn(
+                label: Text('Calories'),
+                tooltip: 'Calories',
+                numeric: true,
+              ),
+              DataColumn(
+                label: Text('Generation'),
+                tooltip: 'Generation',
+              ),
+            ],
+          ),
+        ),
+      );
+    }
+
+    await tester.pumpWidget(MaterialApp(
+      home: buildTable(source),
+    ));
+
+    expect(scrollController.offset, 0.0);
+
+    scrollController.jumpTo(50.0);
+    await tester.pumpAndSettle();
+    expect(scrollController.offset, 50.0);
+  });
+
+  testWidgets('PaginatedDataTable uses PrimaryScrollController when primary ', (WidgetTester tester) async {
+    final ScrollController primaryScrollController = ScrollController();
+    final TestDataSource source = TestDataSource();
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: PrimaryScrollController(
+          controller: primaryScrollController,
+          child: PaginatedDataTable(
+            primary: true,
+            header: const Text('Test table'),
+            source: source,
+            rowsPerPage: 2,
+            columns: const <DataColumn>[
+              DataColumn(label: Text('Name')),
+              DataColumn(label: Text('Calories'), numeric: true),
+              DataColumn(label: Text('Generation')),
+            ],
+          ),
+        ),
+      )
+    );
+
+    // DataTable uses primaryScrollController
+    final Scrollable bodyScrollView = tester.widget(find.byType(Scrollable).first);
+    expect(bodyScrollView.controller, primaryScrollController);
```

Comment:
Adding an additional check like this above should validate the assigned scroll controller is being used.

Reply:
Added this, But for curiosity, would the following be a good solution for this?
```
    // check with position instead of ScrollController's offset to confirm both
    // offset and that the DataTable used ScrollController
    final ScrollPosition position = tester.state<ScrollableState>(find.byType(Scrollable).first).position;

    expect(position.extentBefore, 50.0);  
```
This way both the things get tested at the same time.